core/src/main/java/com/onelogin/saml2/logout/LogoutResponse.java

```diff
@@ -214,20 +224,25 @@ public Boolean isValid(String requestId) {
 				if (cert == null) {
 					throw new SettingsException("In order to validate the sign on the Logout Response, the x509cert of the IdP is required", SettingsException.CERT_NOT_FOUND);
 				}
-
+				
+				
 				String signAlg = request.getParameter("SigAlg");
 				if (signAlg == null || signAlg.isEmpty()) {
 					signAlg = Constants.RSA_SHA1;
 				}
 
-				String signedQuery = "SAMLResponse=" + Util.urlEncoder(request.getParameter("SAMLResponse"));
-
-				String relayState = request.getParameter("RelayState");
-				if (relayState != null && !relayState.isEmpty()) {
-					signedQuery += "&RelayState=" + Util.urlEncoder(relayState);
+				String rawSamlResponse = rawRequestParams.get("SAMLResponse");
```

Comment:
Maybe extract a helper method that extracts the signed query and share that between `LogoutResponse` and `LogoutRequest`